Tinyxml2Serializer.cpp

```diff
@@ -3503,67 +3512,47 @@ std::string CTinyxml2Serializer::SetImageSettings(const unsigned int pCameraId,
 
-    //oXML.AddElem("Plate");
-    //oXML.IntoElem();
+    tinyxml2::XMLElement* forceElem = oXML.NewElement("Force");
+    root->InsertEndChild(forceElem);
 
-    //if (mnMajorVersion > 1 || mnMinorVersion > 7)
-    //{
-    //    AddXMLElementUnsignedInt(&oXML, "Plate_ID", &pPlateId);
-    //}
-    //else
-    //{
-    //    AddXMLElementUnsignedInt(&oXML, "Force_Plate_Index", &pPlateId);
-    //}
-    //if (pCorner1)
-    //{
-    //    oXML.AddElem("Corner1");
-    //    oXML.IntoElem();
-    //    AddXMLElementFloat(&oXML, "X", &(pCorner1->fX));
-    //    AddXMLElementFloat(&oXML, "Y", &(pCorner1->fY));
-    //    AddXMLElementFloat(&oXML, "Z", &(pCorner1->fZ));
-    //    oXML.OutOfElem(); // Corner1
-    //}
-    //if (pCorner2)
-    //{
-    //    oXML.AddElem("Corner2");
-    //    oXML.IntoElem();
-    //    AddXMLElementFloat(&oXML, "X", &(pCorner2->fX));
-    //    AddXMLElementFloat(&oXML, "Y", &(pCorner2->fY));
-    //    AddXMLElementFloat(&oXML, "Z", &(pCorner2->fZ));
-    //    oXML.OutOfElem(); // Corner2
-    //}
-    //if (pCorner3)
-    //{
-    //    oXML.AddElem("Corner3");
-    //    oXML.IntoElem();
-    //    AddXMLElementFloat(&oXML, "X", &(pCorner3->fX));
-    //    AddXMLElementFloat(&oXML, "Y", &(pCorner3->fY));
-    //    AddXMLElementFloat(&oXML, "Z", &(pCorner3->fZ));
-    //    oXML.OutOfElem(); // Corner3
-    //}
-    //if (pCorner4)
-    //{
-    //    oXML.AddElem("Corner4");
-    //    oXML.IntoElem();
-    //    AddXMLElementFloat(&oXML, "X", &(pCorner4->fX));
-    //    AddXMLElementFloat(&oXML, "Y", &(pCorner4->fY));
-    //    AddXMLElementFloat(&oXML, "Z", &(pCorner4->fZ));
-    //    oXML.OutOfElem(); // Corner4
-    //}
-    //oXML.OutOfElem(); // Plate
+    tinyxml2::XMLElement* plateElem = oXML.NewElement("Plate");
+    forceElem->InsertEndChild(plateElem);
 
-    //oXML.OutOfElem(); // Force
-    //oXML.OutOfElem(); // QTM_Settings
+    if (mnMajorVersion > 1 || mnMinorVersion > 7)
+    {
+        AddXMLElementUnsignedInt(plateElem, "Plate_ID", &pPlateId, &oXML);
+    }
+    else
+    {
+        AddXMLElementUnsignedInt(plateElem, "Force_Plate_Index", &pPlateId, &oXML);
+    }
 
-    //return oXML.GetDoc();
+    auto addCorner = [&](const char* name, const SPoint* pCorner)
+        {
+            if (pCorner)
+            {
+                tinyxml2::XMLElement* cornerElem = oXML.NewElement(name);
+                plateElem->InsertEndChild(cornerElem);
 
-    return "";
+                AddXMLElementFloat(cornerElem, "X", &(pCorner->fX), 6, &oXML);
+                AddXMLElementFloat(cornerElem, "Y", &(pCorner->fY), 6, &oXML);
+                AddXMLElementFloat(cornerElem, "Z", &(pCorner->fZ), 6, &oXML);
+            }
+        };
+
+    addCorner("Corner1", pCorner1);
+    addCorner("Corner2", pCorner2);
+    addCorner("Corner3", pCorner3);
+    addCorner("Corner4", pCorner4);
+
+    tinyxml2::XMLPrinter printer;
+    oXML.Print(&printer);
+
+    return printer.CStr();
```

Comment:
Suggestion: I noticed that we are using the explicit constructor in another place. Here we are relying on automatic conversion from char * to std::string using the formal return type. Make sure to be consistent between functions.

Reply:
Removed unnecessary explicit string-construction in all PRs, so it's consistent now.